Tasks/AzureIoTEdgeV2/genconfig.ts

```diff
@@ -0,0 +1,41 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as fs from "fs";
+import * as path from "path";
+import util from "./util";
+import Constants from "./constant";
+import { IExecOptions } from 'azure-pipelines-task-lib/toolrunner';
+
+export async function run() {
+    let templateFilePath: string = tl.getPathInput("templateFilePath", true);
+    tl.debug(`The template file path is ${templateFilePath}`);
+    if (!fs.existsSync(templateFilePath)) {
+        throw Error(tl.loc('TemplateFileInvalid', templateFilePath));
+      }
+      util.setTaskRootPath(path.dirname(templateFilePath));
+    
+      util.setupIotedgedev();
+
+      let envList = {
+        [Constants.iotedgedevEnv.deploymentFileOutputFolder]: tl.getVariable(Constants.outputFileFolder),
+      };
+    
+      // Pass task variable to sub process
+      let tlVariables = tl.getVariables();
+      for (let v of tlVariables) {
+        // The variables in VSTS build contains dot, need to convert to underscore.
+        let name = v.name.replace('.', '_').toUpperCase();
```

Comment:
env could be lower case. why we need make upper case?

Reply:
The build image task converts all envs to upper case, we need to align with build task here

Reply:
Have you tested if I have some lower-cased env in deployment.template.json?

Reply:
Azure Pipelines will convert system variables to environment variables without dot, so we don't need to convert by ourselves.

Reply:
For lower-cased env, all Azure Pipelines environment variables are upper cases, we need to guide users to use upper case variables